projects/packages/assets/src/class-script-data.php

```diff
@@ -69,6 +70,11 @@ public static function register_assets() {
 	 */
 	public static function render_script_data() {
 
+		// Avoid rendering the script data multiple times.
+		if ( current_filter() !== 'enqueue_block_editor_assets' && did_action( 'enqueue_block_editor_assets' ) ) {
```

Comment:
This works, but I wonder if it's more complicated than it needs to be versus having a `private static $did_render_script_data = false;` on the class that gets checked here and then set true.

Reply:
IMHO, this conveys more information about why we have an early return here and it tells you exactly in which case the data can be duplicated.

Reply:
On the other hand, if we have to change which hooks are used then this might wind up being obsolete. But like I said, I'm ok if you go ahead with this as it is. 🙂

Reply:
Alright, I have used the static variable and added a comment about when it can be called multiple times. Thanks